src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1308,6 +1311,8 @@ int wm_vuldet_send_agent_report(sqlite3 *db, OSHash *cve_table, agent_software *
                 }
             }
 
+            //Save the vulnerability in the agent database
+            wdb_agents_vuln_cve_insert(atoi(report->agent_id), report->software, report->version, report->arch, report->cve, &sock);
```

Comment:
We could check the return value of the function, to print a message for example.
Also, if the process fails, we could even clear the table to avoid partial or incomplete results.

Reply:
Fixed